src/App.js

```diff
@@ -1,26 +1,120 @@
 import React from 'react';
-import logo from './logo.svg';
 import './App.css';
+import Header from './components/Header';
+import Footer from './components/Footer';
+import Main from './components/Main';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      value: '',
+      items: [],
+      selectedList: 0,
+      checked: false,
+    };
+
+    this.addToValue = (event) => {
+      this.setState({
+        value: event.target.value,
+      });
+    };
+
+    this.addToItems = () => {
+      if (this.state.value) {
+        this.setState(prevState => ({
+          items: [
+            ...prevState.items,
+            {
+              value: this.state.value,
+              done: false,
+            },
+          ],
+
+          value: '',
+        }));
+      }
+    };
+
+    this.handleClickToDone = (value) => {
+      for (const key of this.state.items) {
+        if (key.value === value) {
+          key.done = !key.done;
```

Comment:
--- OBJECT MUTATION

Do you remember what happens when you try to assign object to the variable?
```javascript
const a = {
  name: 'Vasya'
  age: 21
}
const b = a;
const c = b;
b.age = 22;
c.name = 'Petya';
console.log(a); // {name: 'Petya', age: 22}
console.log(b); // {name: 'Petya', age: 22}
console.log(c); // {name: 'Petya', age: 22}
```
You should avoid this kind of code. Use `...spread` operator to create a copy of object and override needed fields.

Reply:
Also, it is not safe and optimized to go through all items. Imagine you have a couple of todo items with the same value. When you click on the one of them, the second will be toggled along with clicked. When you add `ID`, it will be much easier to change needed item. Let's consider you passed `id` instead of `value`.
Suggestion:
```javascript
// first - copy an array of items
const items = [...this.state.items];

// find needed item index
const itemIndex = items.findIndex((item) => item.id === id);

// toggle done status
items[itemIndex] = {
  ...items[itemIndex],
  done: !items[itemIndex].done,
}

// set new state
this.setState({ items })
```